src/Resources/app/administration/test/e2e/cypress/integration/plugin-config.spec.js

```diff
@@ -0,0 +1,186 @@
+            text: "Choose a product for the plugin configuration",
+        },
+        elementSelector: 'div[name="PluginConfig.config.exampleProduct"]',
+    },
+    {
+        type: "multi-select",
+        options: ["Cool Product"],
+        label: {
+            selector: 'label[for="PluginConfig.config.exampleMultiProductIds"]',
+            text: "Choose multiple products IDs for the plugin configuration",
+        },
+        elementSelector:
+            ".sw-entity-multi-select > .sw-block-field__block > .sw-select__selection",
+    },
+    {
+        type: "media",
+        label: {
+            selector: 'div[name="PluginConfig.config.pluginMedia"]>label',
+            text: "Upload media or choose one from the media manager",
+        },
+        elementSelector:
+            'div[name="PluginConfig.config.pluginMedia"]>.sw-media-field__input-container',
+    },
+    {
+        type: "html-editor",
+        label: {
+            selector:
+                'div[name="PluginConfig.config.textEditor"]>.sw-text-editor__label',
+            text: "Write some nice text with WYSIWYG editor",
+        },
+        elementSelector:
+            "div.sw-text-editor div.sw-text-editor__content-editor",
+    },
+];
+
+describe("Administration: Plugin Config", () => {
+    beforeEach(() => {
+        return cy
+            .loginViaApi()
+            .then(() => {
+                return cy.setLocaleToEnGb();
+            })
+            .then(() => {
+                try {
+                    cy.createProductFixture();
+                } catch (_) {
+                    // This allows the test to be run multiple times without deleting the previous product.
+                    // If the product is not created successfully the test will fail anyway
+                }
+            });
+    });
+
+    it("should be able to fill all config fields", () => {
+        cy.visit("/admin#/sw/plugin/index/list");
+        // Open the Context Menu of the plugin
+        cy.get(".sw-card__content")
+            .contains("Label for the plugin PluginConfig")
+            .parents("tr")
+            .find(".sw-context-button__button")
+            .click();
+        // Click the context button
+        cy.get(".sw-context-menu")
+            .should("be.visible")
+            .contains("Config")
+            .should("be.visible")
+            .click();
+        cy.url().should("include", "/admin#/sw/plugin/settings/PluginConfig");
+
+        for (const item of configElements) {
+            cy.log(`Testing ${item.type}`);
+            cy.contains(item.label.selector, item.label.text);
+            cy.get(item.elementSelector).scrollIntoView().should("be.visible");
+
+            switch (item.type) {
+                case "input":
+                    cy.get(item.elementSelector).clearTypeAndCheck(
+                        "test@shopware.com",
+                    );
+                    break;
+                case "single-select":
```

Comment:
Ich find dieses for/switch/case konstrukt nicht so cool. Mach doch mehrere Tests dafür. Dann brauchst du auch nicht bangen dass es wegen statischem code failed.

Reply:
Ich seh gerade du hast nur genau einen case. Warum so umständlich?

schau mal wie ich das bei cms-extensions gemacht habe im `scroll-navigation.spec`:

```js
    context('shows navigation menu/sidebar when visiting the landing page', () => {
        contexts.forEach((context) => {
            it(`on ${context.deviceType}`, () => {
```
So machst du aus den einzelnen Dingen cases und das ist dann das einzige was du iterierst.

Reply:
Hab das alles jetzt aufgeteilt